src/snarky.d.ts

```diff
@@ -499,6 +496,8 @@ export class Circuit {
   // static runAndCheck<T>(f : () => Promise<(() => T)>): Promise<T>;
   static runAndCheck<T>(f: () => Promise<() => T>): Promise<T>;
 
+  static runAndCheckSync<T>(f: () => T): T;
```

Comment:
when would you use `runAndCheckSync` vs `runAndCheck`? Is this for web-workers?

Reply:
runAndCheck was an async version made by Izzy, but it caused some bugs, like this one: https://github.com/o1-labs/snarkyjs/issues/65
I don't think the async version is really fully compatible with how `snarky` works right now. At some point, I couldn't get something to work with the async version, so I added the sync version. The async version is not used anymore, should probably just replace it with the sync version.

Reply:
Can you delete the async version from the code so people don't accidentally try to use it?

Reply:
done https://github.com/o1-labs/snarkyjs/pull/123/commits/2a9c15e2568e114405e652ef3cb9e82048c17d45